library/sha1.c

```diff
@@ -310,7 +310,7 @@ int mbedtls_sha1_update_ret( mbedtls_sha1_context *ctx,
     uint32_t left;
 
     MBEDTLS_SHA1_VALIDATE_RET( ctx != NULL );
-    MBEDTLS_SHA1_VALIDATE_RET( ilen == 0 || input != NULL );
+    MBEDTLS_SHA1_VALIDATE_RET( ilen != 0 || input != NULL );
```

Comment:
Is this change in behavior correct?

Reply:
No, that's rubbish, thanks for spotting - I amended the commit.